btag.py

```diff
@@ -15,15 +15,35 @@
 # https://btv-wiki.docs.cern.ch/PerformanceCalibration/SFUncertaintiesAndCorrelations/working-point-based-sfs-fixedwp-sfs
 # https://btv-wiki.docs.cern.ch/PerformanceCalibration/fixedWPSFRecommendations/
 
+
+# shouldcorrectly process source name containing two underscores (e.g. JES_Absolute_2022)
+# will correctly cut out second word and check if its in the list
+def InList(src_name, jes_list):
+    split_src_name = src_name.split('_')
+    if len(split_src_name) == 1:
+        return src_name in jes_list
+    elif len(split_src_name) > 1:
+        to_check = split_src_name[0] + '_'
+        return to_check in jes_list
+    else:
+        raise RuntimeError(f"Cannot parse source name: {src_name}")
+
+
 class bTagCorrProducer:
     jsonPath = "/cvmfs/cms.cern.ch/rsync/cms-nanoAOD/jsonpog-integration/POG/BTV/{}/btagging.json.gz"
     bTagEff_JsonPath = "Corrections/data/BTV/{}/btagEff.root"
     initialized = False
     uncSource_bTagWP = ["btagSFbc_uncorrelated", "btagSFlight_uncorrelated","btagSFbc_correlated", "btagSFlight_correlated"]
-    uncSources_bTagShape_shape = ["FlavorQCD","RelativeBal", "HF", "BBEC1", "EC2", "Absolute", "BBEC1_", "Absolute_", "EC2_", "HF_", "RelativeSample_" ]
+    uncSources_bTagShape_jes = ["Total", "FlavorQCD","RelativeBal", "HF", "BBEC1", "EC2", "Absolute", "BBEC1_", "Absolute_", "EC2_", "HF_", "RelativeSample_" ]
     uncSources_bTagShape_norm = ["lf", "hf", "lfstats1", "lfstats2", "hfstats1", "hfstats2", "cferr1", "cferr2"]
 
-    def __init__(self, period, loadEfficiency=True):
+    tagger_to_brag_branch = {"particleNet": "PNetB",
+                             "deepJet": "DeepFlavB"}
+
+    def __init__(self, period, loadEfficiency=True, tagger_name="particleNet", use_split_jes=False):
```

Comment:
pass `tagger_name` from config. Use `deepJet` as the default value to be back compatible with Run 2